### DISCUSSION TITLE
running on selected cells

### DISCUSSION THREAD

**Discussion by aminzia** [theislab/scvelo#461] [Q&A]
Hello everyone, 

I was wondering if there's a way to run scvelo on a selected set of 10X gex cells (a subset of barcodes). 

Thank you all in advance.
Amin

**Comment by WeilerP** [comment]
Currently, there is no such option in `scvelo`. I guess you were hoping for some keyword argument like `var_names` in `scv.tl.recover_dynamics`. The reason this is not as easily possible is that subsetting observations requires also recalculating the neighbor graph and subsequently imputation and latent space embedding.
To summarise: To run scvelo on a subset of your observations, you will have to subset the data manually:

```python
adata_subsetted = adata[obs_subset, :]

# preprocessing
scv.pp.moments(adata_subsetted)

# inference
scv.tl.recover_dynamics(adata_subsetted)
```
